File: frontend/src/utils/schemaEditor/validate.ts
import { DatabaseEdit, ValidateResult } from "@/types";

export const validateDatabaseEdit = (
  databaseEdit: DatabaseEdit
): ValidateResult[] => {
  const validateResultList: ValidateResult[] = [];

  for (const createTableContext of databaseEdit.createTableList) {
    if (createTableContext.addColumnList.length === 0) {
      validateResultList.push({
        type: "ERROR",
        message: `Table ${createTableContext.name} should has at least one column.`,
      });
    }
  }

  return validateResultList;
};

File: backend/plugin/parser/edit/pg/validate.go
package pg

import (
	api "github.com/bytebase/bytebase/backend/legacyapi"
)

// ValidateDatabaseEdit validates the api message DatabaseEdit, including related column type.
func (*SchemaEditor) ValidateDatabaseEdit(_ *api.DatabaseEdit) ([]*api.ValidateResult, error) {
	// Because Postgres has a custom data type, so skip directly here.
	return []*api.ValidateResult{}, nil
}
